modules/internal/ChapelArray.chpl

```diff
@@ -19,6 +19,118 @@
+
+     arr1 == arr2 // compare each element resulting in an array of booleans
+     arr1 != arr2 // compare each element resulting in an array of booleans
+     arr1.equals(arr2) // compare entire arrays resulting in a single boolean
+
+   ========================================
+   Miscellaneous Domain and Array Operators
+   ========================================
+
+   The domain count operator ``#``
+   -------------------------------
+
+   The ``#`` operator can be applied to dense rectangular domains
+   with a tuple argument whose size matches the rank of the domain
+   (or optionally an integer in the case of a 1D domain). The operator
+   is equivalent to applying the ``#`` operator to the component
+   ranges of the domain and then using them to slice the domain.
+
+   The array count operator ``#``
+   ------------------------------
+   The ``#`` operator can be applied to dense rectangular arrays
+   with a tuple argument whose size matches the rank of the array
+   (or optionally an integer in the case of a 1D array). The operator
+   is equivalent to applying the ``#`` operator to the array's domain
+   and using the result to slice the array.
+
+   The array swap operator ``<=>``
+   -------------------------------
+   The ``<=>`` operator can be used to swap the contents of two arrays
+   with the same shape.
+
+   The array alias operator ``=>``
+   -------------------------------
+
+   The ``=>`` operator can be used in a variable declaration to create
+   a new alias of an array. The new variable will refer to the same
+   array elements as the aliased array.  In the following example,
+   the variable ``Inner`` refers to the inner 9 elements of ``A``.
+
+   .. code-block:: chapel
+
+     var A: [0..10] int;
+     var Inner => A[1..9];
+
+   ================================================
+   Set Operations on Associative Domains and Arrays
+   ================================================
+
+   Associative domains and arrays support a number of operators for
+   set manipulations.  The supported set operators are:
+
+     =====  ====================
+     \+ \|    Union
+     &      Intersection
+     \-      Difference
+     ^      Symmetric Difference
+     =====  ====================
+
+   Consider the following code where ``A`` and ``B`` are associative arrays:
+
+   .. code-block:: chapel
+
+     var C = A op B;
+
+   The result ``C`` is a new associative array backed by a new associative
+   domain. The domains of ``A`` and ``B`` are not modified by ``op``.
+
+   There are also op= variants that store the result into the first operand.
+   
+   Consider the following code where ``A`` and ``B`` are associative arrays:
+
+   .. code-block:: chapel
+
+     A op= B;
+
+   ``A`` must not share its domain with another array, otherwise the program
+   will halt with an error message.
+
+   For the ``+=`` and ``|=`` operators, the value from ``B`` will overwrite
+   the existing value in ``A`` when indices overlap.
```

Comment:
Might be useful to suggest that the user look at the sets readme for more information.